pkg/updater/version.go

```diff
@@ -61,7 +64,45 @@ type AppResponse struct {
 }
 
 type versionConfig struct {
-	ServerURL string `env:"ABC_UPDATER_URL,default=https://abc-updater.tycho.joonix.net"`
+	ServerURL      string   `env:"ABC_UPDATER_URL,default=https://abc-updater.tycho.joonix.net"`
+	IgnoreVersions []string `env:"IGNORE_VERSIONS"`
+}
+
+func (c *versionConfig) ignoreAll() bool {
+	for _, version := range c.IgnoreVersions {
+		if strings.ToLower(version) == "all" {
+			return true
+		}
+	}
+	return false
+}
+
+// IsIgnored returns true if the version specified should be ignored.
+func (c *versionConfig) isIgnored(checkVersion string) (bool, error) {
+	v, err := version.NewVersion(checkVersion)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse version: %w", err)
+	}
+
+	var cumulativeErr error
+	for _, ignoredVersion := range c.IgnoreVersions {
+		if strings.ToLower(ignoredVersion) == "all" {
```

Comment:
It feels a little weird that this `"all"` logic is kinda duplicated in both `isIgnored()` and `ignoreAll()`. Maybe isIgnored() should call `if c.ignoreAll()` at the top, and then skip the check for "all" in this loop? It's one more iteration over the list, but we can spare a few microseconds here.

Reply:
Yeah, I had done it to avoid a 2nd loop but I guess that would be cleaner